test/records/test_echo_interface.py

```diff
@@ -103,3 +143,80 @@ def test_upload_file_object(self, _):
 
         test_echo.upload_file_object(BytesIO(), self.test_image_path)
         assert test_echo.bucket.upload_fileobj.call_count == 1
+
+        with patch("boto3.resource") as mock_resource:
+            mock_bucket = mock_resource.return_value.Bucket.return_value
+
+            mock_bucket.upload_fileobj.side_effect = ClientError(
+                {"Error": {"Code": "AccessDenied", "Message": "Access Denied"}},
+                "upload_fileobj",
+            )
+
+            test_echo = EchoInterface()
+
+            with pytest.raises(EchoS3Error, match="when uploading file at"):
+                test_echo.upload_file_object(BytesIO(), self.test_image_path)
```

Comment:
Similar to above, as this is testing for things going wrong, this should be in a separate test function. Rename this one to test_valid_upload_file_object() and create a new function called test_invalid_upload_file_object().

Reply:
resolved in next commit